### DISCUSSION TITLE
Why even have secret levels?

### DISCUSSION THREAD

**Discussion by Pheazant** [SuperTux/supertux#2179] [Ideas]
Player curiosity and bragging rights aside.

World 2's secret tower might be once of the hardest levels in the main campaign so far. What reward should the player get for besting it?

Here are a few possibilities:

 - Fast travel points/shortcuts in the worldmap
 - A strong powerup exclusive to secret levels (possibly the Earth Flower from #1820?)
 - The secret-based shop currency mentioned in the [design document](https://github.com/SuperTux/supertux/wiki/Milestone-3-Design-Document#shop)
 - Shops themselves
 - Something related to the scrapped keys mechanic? Four keys (plus one final) map to the four worlds pretty well.

Has this been discussed before?

**Comment by Semphriss** [comment]
If the secret tower refers to the Tower of Ghosts, this level won't stay. It has been acknowledged for quite a while now that it wasn't well designed (excessively difficult, enemy spam, not enough organization and more) and will be replaced whenever the team will have finished its replacement.

**Reply by Pheazant** [reply]
I re-checked - Ancient Ruins was the level I was talking about. It's not necessary for the main plot's progression and quite protracted, too. Its complexity is not the issue, though. I am okay with a hard bonus level - but the pleasure of checking off a secret map point shouldn't be treated as a reward on its own.

**Reply by Semphriss** [reply]
> [...] the pleasure of checking off a secret map point shouldn't be treated as a reward on its own.

Disagreement from about all the players I can think of who gave their opinion in various designing sessions. Fun secrets are part of what makes the game fun and non-linear; it gives a motivation to seek for secret exits.

**Reply by Pheazant** [reply]
I see. Cutting the bonus levels altogether is a bit of a strawman point - I also am a curious player as well, they are a fun concept.

Compare that to the Super Mario World's secret design, though. Almost none of the alternate paths end in a dead end, and the ones that do provide the player with something unique for finding them.

The way the first world's bonus has had a history of showing off content to come in future versions is really nice, for one. It's like a secret developer room - albeit unsustainable once the game runs out of content to be added. Ancient Ruins should have something going for it, too - something to keep the player unlocking it on runs after the first one.

**Comment by provigz** [comment]
In my opinion, secret levels give some variety of fun things to explore and look for. It's usually a pleasant feeling for the player when they discover one. Sure, they may not be important for the completion of a given map (in this case Story Mode), but I think they make as cool bonuses to look for and enjoy.

**Comment by Anonymous941** [comment]
Between Two Glaciers has crystals that could be used to have secret stuff, but for now they don't do anything.  Perhaps there could be an entire secret world (like Super Mario World's special zone and star world) that clearing has an effect on the game?
